### ISSUE TITLE
Walk location is wrong every second time in room 2

### ISSUE THREAD

**Issue by balloonpopper** [godot-escoria/escoria-issues#295]
**Describe the bug**
Start in room 3 and walk into room 2. 
Click somewhere on the left hand platform. Continue to click that spot without moving your mouse between clicks. The player alternates between walking to the nearest point they can get to, and walking to the right hand door.

**To Reproduce**
As above

**Expected behavior**
The player should always walk to the closest point to your click point.

**Screenshots**

**Versions**
* Escoria: Latest develop
* Godot: 3.5

**Additional context**

**Comment by BHSDuncan** [comment]
This seems to be something caused by `get_simple_path()`. Will take a look into it.

**Comment by BHSDuncan** [comment]
https://docs.godotengine.org/en/3.4/classes/class_navigation.html would seem to indicate there are issues, although they aren't specific about it in that link. `Navigation` has also been deprecated, but that's only as of Godot 3.5 and its replacement also isn't available before 3.5.

@StraToN how did you want to proceed with this?

**Comment by StraToN** [comment]
Since Navigation will deprecated soon we'll have to move out of it sooner or later. 
This can wait after Escoria 4.0 release though, but I'd have this done for 4.1 at most.

**Comment by BHSDuncan** [comment]
Fair enough, but maybe we should probably discuss what approach we're going to take seeing as the `Navigation` replacement is new to Godot 3.5.
